File: backend_project_centropsicologico_pisw2025-main/mer/src/main/java/com/sensespsicologos/mer/models/PatientTest.java
package com.sensespsicologos.mer.models;

import com.sensespsicologos.mer.common.models.BaseEntity;
import jakarta.persistence.*;
import lombok.*;

import java.time.Instant;

@Entity
@Table(name = "patient_tests")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class PatientTest extends BaseEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "test_id")
    private Test test;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "clinical_history_id")
    private ClinicalHistory clinicalHistory;

    @Builder.Default
    @Column(name = "is_general_doc", nullable = false)
    private Boolean isGeneralDoc = false;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "completed_by_id")
    private User completedBy;

    @Column(name = "completed_at", nullable = false)
    private Instant completedAt;

    @OneToOne(fetch = FetchType.LAZY, cascade = CascadeType.ALL)
    @JoinColumn(name = "document_id", referencedColumnName = "id")
    private Document document;

    @PrePersist
    protected void onCreate() {
        if (completedAt == null) {
            completedAt = Instant.now();
        }
    }
}

File: backend_project_centropsicologico_pisw2025-main/src/interfaces/PatientTest.ts
import { Appointment } from "./Appointment";
import { ClinicalHistory } from "./ClinicalHistory";
import { Document } from "./Document";
import { Test } from "./Test";
import { User } from "./User";

export interface PatientTest {
  id: string;

  testId: string;
  test: Test;

  clinicalHistoryId: string;
  clinicalHistory: ClinicalHistory;

  appointmentId?: String;
  appointment?: Appointment;

  isGeneralDoc: boolean;

  document?: Document;

  completedById: string;
  completedBy: User;

  completedAt: Date;
}
